sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/implementation/MultipartFormDataHelperTest.java

```diff
@@ -0,0 +1,161 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Unit tests for MultipartFormDataHelper.
+ */
+public class MultipartFormDataHelperTest {
+    @Test
+    public void testSerializeJsonField() {
+        // Arrange
+        RequestOptions multipartRequestOptions = new RequestOptions();
+
+        // Act
+        MultipartFormDataHelper multipartRequest
+            = new MultipartFormDataHelper(multipartRequestOptions).serializeJsonField("weather", "sunny")
+            .serializeJsonField("temperatureCelsius", 24);
+
+        BinaryData requestBody = multipartRequest.end().getRequestBody();
+
+        // Assert
+        assertTrue(requestBody.toString().contains("Content-Disposition: form-data; name=\"weather\""));
+        assertTrue(requestBody.toString().contains("Content-Disposition: form-data; name=\"temperatureCelsius\""));
+        assertTrue(requestBody.toString().contains("Content-Type: application/json"));
+        assertTrue(requestBody.toString().contains("sunny"));
+        assertTrue(requestBody.toString().contains("24"));
+    }
+
+    @Test
+    public void testSerializeJsonFieldNull() {
+        // Arrange
+        RequestOptions multipartRequestOptions = new RequestOptions();
+
+        // Act
+        MultipartFormDataHelper multipartRequest
+            = new MultipartFormDataHelper(multipartRequestOptions).serializeJsonField("weather", null);
+
+        BinaryData requestBody = multipartRequest.end().getRequestBody();
+
+        // Assert
+        assertFalse(requestBody.toString().contains("Content-Disposition: form-data; name=\"weather\""));
+        assertFalse(requestBody.toString().contains("Content-Type: application/json"));
+    }
+
+    @Test
+    public void testSerializeFileFields() {
+        // Arrange
+        RequestOptions multipartRequestOptions = new RequestOptions();
+        MultipartFormDataHelper multipartRequestOne
+            = new MultipartFormDataHelper(multipartRequestOptions).serializeJsonField("weather", "sunny");
+        MultipartFormDataHelper multipartRequestTwo
+            = new MultipartFormDataHelper(multipartRequestOptions).serializeJsonField("weather", "rainy");
+        MultipartFormDataHelper multipartRequestThree
+            = new MultipartFormDataHelper(multipartRequestOptions).serializeJsonField("weather", "cloudy");
+        BinaryData binaryDataOne = multipartRequestOne.end().getRequestBody();
+        BinaryData binaryDataTwo = multipartRequestTwo.end().getRequestBody();
+        BinaryData binaryDataThree = multipartRequestThree.end().getRequestBody();
+
+        List<BinaryData> files = new ArrayList<>();
+        files.add(binaryDataOne);
+        files.add(binaryDataTwo);
+        files.add(binaryDataThree);
+
+        List<String> contentTypes = new ArrayList<>();
+        contentTypes.add(null);
+        contentTypes.add("");
+        contentTypes.add("application/mocktype");
```

Comment:
Nit: using `Arrays.asList(...)` is a bit more concise.
```suggestion
        List<String> contentTypes = Arrays.asList(
        null,
        "",
        "application/mocktype");
```

You will probably have to add the import for `Arrays`